hwbench/environment/block_devices.py

```diff
@@ -0,0 +1,132 @@
+import pathlib
+from json import JSONDecodeError, loads
+from typing import Any
+
+import pyudev
+
+from hwbench.utils import helpers as h
+from hwbench.utils.external import External
+
+
+class Block_Device:
+    """Block_Device is a class that gathers block_device information"""
+
+    def __init__(self, out_dir, udev_device):
+        self.udev_device = udev_device
+        self.name = udev_device.sys_name
+        self.type = self.get_block_device_type()
+        self.out_dir = out_dir
+        self.smart_json = b"{}"
+
+    def is_rotational(self) -> bool:
+        syspath = f"/sys/block/{self.udev_device.sys_name}/queue/rotational"
+        with open(syspath) as file:
+            content = file.read()
+        return content.lower() in ("yes", "true", "t", "1")
+
+    def is_ata(self) -> bool:
+        bus = self.udev_device.get("ID_BUS")
+        return bool(bus == "ata")
+
+    def is_nvme(self) -> bool:
+        return bool(self.udev_device.parent.get("NVME_TRTYPE"))
+
+    def get_block_device_type(self) -> str:
```

Comment:
Thinking of this one, I think this should be an Enum type with an str method to get the string.
That will make it easier to be reused, especially with hwgraph.

Something like : 
class blockdevicetype(Enum):
  HDD = 'hdd'
  SSD = 'ssd'
  UNKNOWN = 'unknown'

 def __str__(self):
    return str(self.value)

Reply:
good idea. done.